airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py

```diff
@@ -176,23 +176,18 @@ def dict(self, *args, **kwargs):
         return super().dict(*args, **kwargs)
 
 
-def split_config(config: Mapping[str, Any]) -> Tuple[dict, InternalConfig]:
+def split_config(config: MutableMapping[str, Any]) -> Tuple[MutableMapping, InternalConfig]:
```

Comment:
Here the use case of popping items from the original config makes sense if the object passed to it is indeed always loaded from the config file, but since we can't validate that from within the function, I wonder if this could lead to unexpected usage elsewhere re: changing the original dict. Maybe renaming to something like `filter_internal_keywords` would make it clearer both what happens in the mutation (i.e. if you were to call it without using the return values) as well as what the return values are

Reply:
My only comment -- otherwise I took a look, really cool stuff 👀

Reply:
👍 Yes you're right! I renamed the function to `filter_internal_keywords`